docs/decisions/0054-processes.md

```diff
@@ -0,0 +1,317 @@
+
+```
+
+The two steps steps created above have been added to our new `ChatBot` process and the `UserInputStep` has been declared as the entry point. This means that any events received by the process will be forwarded to this step. Now we need to define the flow of our process by describing which actions are triggered by events from our steps.
+
+```csharp
+
+// When the userInput step completes, send the output to the llm response step
+userInputStep
+    .OnFunctionResult(nameof(UserInputStep.GetUserInput))
+    .SendOutputTo(responseStep, nameof(ChatBotResponseStep.GetChatResponse), "userMessage");
+
+```
+
+In the code above, `userInputStep.OnFunctionResult(nameof(UserInputStep.GetUserInput))` selects the event that is emitted by the process on successful execution of the `GetUserInput` KernelFunction in the step instance referenced by `userInputStep`. It then returns a builder type object that provides actions based on the context. In this case the `SendOutputTo(responseStep, nameof(ChatBotResponseStep.GetChatResponse), "userMessage")` action is used to forward the event data to the `userMessage` parameter of the `GetChatResponse` KernelFunction on the step instance referenced by `responseStep`.
+
+One of the key takeaways here is that events emitted by a given step can be selected and forwarded to **_a specific parameter of a specific KernelFunction_** within another step. Event data sent to parameters of KernelFunctions are queued until all of the required parameters of the function have received input, at which point the function will be invoked.
+
+#### Step 3 - Get output from the Process:
+
+Now that we've defined our process, we would like to inspect the final result that it produces. In many cases the result of the process will be written to a database or queue or some other internal system and that's all that's needed. In some cases however, such as in the case of a process running in a server as the result of a synchronous REST call, there is a need to extract the result from the finished process so that it can be returned to the caller. In these cases handler functions can be registered on the process to be triggered by a specific event.
+
+Let's wire up the process above to run a handler function when the `ChatBotResponseStep` step completes.
+
+```csharp
+
+process.OnEvent(ChatBotEvents.AssistantResponseGenerated).Run((CloudEvent e) =>
+{
+    result = (int)e.Data!;
+    Console.WriteLine($"Result: {result}");
+});
+
+```
+
+A key thing to notice is that the event emitted by the `ChatBotResponseStep` within the processes was also be emitted from the processes itself which allows us to register a handler for it. All events within a process will bubble up out of the process to the parent which may be the program running the process or may be another process. This pattern allows for nested processes where an existing process can be used as a step in another process.
+
+#### Step 4 - Process object model:
```

Comment:
Is there a requirement that processes should be transferable between SK languages?

Reply:
IMHO as a developer I will use a single language and stick to it... homogeneity is crucial... other thing is that the in-memory construct and the set of instructions match in all languages, then theoretically it could be ported or have a compatibility layer... 

Just thinking loud without having seen the internals, so might be talking nonsense :P